### PR TITLE
Problem: no option to exit early for fullnodes

### PR DESCRIPTION
**Pull request by mmsqe**
add exit-on-finish to allow early quit after send txs are done

👮🏻👮🏻👮🏻 !!!! REFERENCE THE PROBLEM YOUR ARE SOLVING IN THE PR TITLE AND DESCRIBE YOUR SOLUTION HERE !!!! DO NOT FORGET !!!! 👮🏻👮🏻👮🏻

# PR Checklist:

- [ ] Have you read the [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md)?
- [ ] Does your PR follow the [C4 patch requirements](https://rfc.zeromq.org/spec:42/C4/#23-patch-requirements)?
- [ ] Have you rebased your work on top of the latest master? 
- [ ] Have you checked your code compiles? (`make`)
- [ ] Have you included tests for any non-trivial functionality?
- [ ] Have you checked your code passes the unit tests? (`make test`)
- [ ] Have you checked your code formatting is correct? (`go fmt`)
- [ ] Have you checked your basic code style is fine? (`golangci-lint run`)
- [ ] If you added any dependencies, have you checked they do not contain any known vulnerabilities? (`go list -json -m all | nancy sleuth`)
- [ ] If your changes affect the client infrastructure, have you run the integration test?
- [ ] If your changes affect public APIs, does your PR follow the [C4 evolution of public contracts](https://rfc.zeromq.org/spec:42/C4/#26-evolution-of-public-contracts)?
- [ ] If your code changes public APIs, have you incremented the crate version numbers and documented your changes in the [CHANGELOG.md](https://github.com/crypto-org-chain/chain-main/blob/master/CHANGELOG.md)?
- [ ] If you are contributing for the first time, please read the agreement in [CONTRIBUTING.md](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://github.com/crypto-org-chain/chain-main/blob/master/CONTRIBUTING.md#developer-certificate-of-originn).

Thank you for your code, it's appreciated! :)

## Summary by CodeRabbit

- **New Features**
	- Introduced a new command-line option `--exit-on-finish` for enhanced control over process completion behavior.
  
- **Improvements**
	- Updated Docker image tag to a specific version for improved stability and predictability in deployments.
	- Reduced the number of service instances from 10 to 8, optimizing resource allocation and testing strategy.

### CODE REVIEW

File: testground/benchmark/benchmark/stateless.py [hunk 1]
```diff
@@ -225,7 +235,8 @@ def do_run(home: str, cronosd: str, group: str, global_seq: int, cfg: dict):
         asyncio.run(send_txs(txs))
 
     # node quit when the chain is idle or halted for a while
-    detect_idle_halted(20, 20)
+    if group == VALIDATOR_GROUP or (group == FULLNODE_GROUP and not exit_on_finish):
```
Comment (yihuang) [inline testground/benchmark/benchmark/stateless.py:238]: the problem with early quit is it'll lose lots of transactions that are in the fullnode's mempool but not propagated to the validator yet.

Comment (mmsqe) [reply testground/benchmark/benchmark/stateless.py:238]: yea, quite tricky in local since validator also slows down when fullnode survives

Comment (yihuang) [reply testground/benchmark/benchmark/stateless.py:238]: with validator-generate-load, can you handle a single full node, or you want all full nodes gone? currently only a single full node is required for generating genesis, but that can be changed to validator too.

Comment (mmsqe) [reply testground/benchmark/benchmark/stateless.py:238]: I think so if you mean allow no fullnode as well

Comment (yihuang) [reply testground/benchmark/benchmark/stateless.py:238]: we can certainly do that

**Comment by yihuang** [comment]
closing because we'll support validator only mode.

**Pull request closed without merging**